assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js

```diff
@@ -124,11 +130,21 @@ export default function PublicationApprovedOverlayNotification() {
 
 	// In useEffect, set publicationOnboardingStateChanged to false using setPublicationOnboardingStateChanged method and save the setting using saveSettings action. This effect should be run only once when component is mounted.
 	useEffect( () => {
-		if ( initialPublicationOnboardingStateChanged.current === true ) {
-			setPublicationOnboardingStateChanged( false );
-			saveSettings();
+		if ( initialPublicationOnboardingStateChanged.current === undefined ) {
+			initialPublicationOnboardingStateChanged.current =
+				publicationOnboardingStateChanged;
+
+			if ( publicationOnboardingStateChanged === true ) {
+				setPublicationOnboardingStateChanged( false );
+				saveSettings();
+			}
 		}
-	}, [ saveSettings, setPublicationOnboardingStateChanged ] );
+	}, [
+		hasSettingsResolved,
```

Comment:
I see `hasSettingsResolved` is a `useEffect` dependency but I don't see that used within the callback. Is this intentional? Do we only want to process the callback only when `hasSettingsResolved` is truthy?

Reply:
Thanks @nfmohit , yes, we need to run this effect when settings have been resolved. I've updated the condition inside `useEffect`.